### PR TITLE
#161115790 Add test for Auth module

### PR DESCRIPTION
**Pull request by veeqtor**
#### What does this PR do?
    -  Adds test for Auth module

#### Description of Task to be completed?
    - write tests for auth component
    - write tests for auth container

#### How should this be manually tested?
  - run the `git pull` command
  - run `npm test`
  - the coverage report should show that modules/Auth has 100% coverage

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161115790](https://www.pivotaltracker.com/story/show/161115790)

#### Screenshots or gifs (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: client/src/tests/modules/auth/container/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      {...propsNew} />);
+    mockWrapper.instance().handleSuccess(responses.success);
+    expect(spy).toHaveBeenCalled();
+  });
+
+
+  it('Should call the handleSuccess method', () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().handleSuccess(responses.error);
+    expect(wrapper).toMatchSnapshot();
```
Comment (chidioguejiofor) [inline client/src/tests/modules/auth/container/index.test.jsx]: This function calls state and sets the `data` property of the state object. What do you think about testing that the state is correctly set here as opposed to testing `snapshot`

File: client/src/tests/modules/auth/container/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+
+  it('Should call the handleFailure method', () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().handleFailure(responses.error);
+    const actual = wrapper.instance().state.showForm;
+    expect(actual).toEqual(false);
```
Comment (chidioguejiofor) [inline client/src/tests/modules/auth/container/index.test.jsx]: Nice

File: client/src/tests/modules/auth/container/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsErrors);
+    const actual = wrapper.instance().state;
+    expect(actual).toEqual(expected.notSuccessful);
+  });
+
+
+  it("should test for returned messages on authentication ", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsMessage);
```
Comment (chidioguejiofor) [inline client/src/tests/modules/auth/container/index.test.jsx]: what do you think about using the `wrapper.setProps(nextProps: object)` function to test componentWillReceiveProps as opposed to calling it directly like this

File: client/src/tests/modules/auth/container/index.test.jsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+
+  it("should call the complete signUp method", () => {
+    const spy = jest.fn();
+    const propsNew = { signUp: spy };
+    const mockWrapper = shallow(<SignIn
+      {...props.containerProps}
+      {...propsNew} />);
+    mockWrapper.instance().completeSignUp({ preventDefault: jest.fn() });
+    expect(spy).toHaveBeenCalled();
```
Comment (chidioguejiofor) [inline client/src/tests/modules/auth/container/index.test.jsx]: clean

**Review by chidioguejiofor — COMMENTED** [review]
Nice work on this one. Found loads of clean code here. 
To add to the few suggestions that were made on your files, I want to know what you think about adding a screenshot of the files that you tested showing their test coverage?

File: client/src/tests/modules/auth/authMockData.js [hunk 5]
```diff
@@ -1,11 +1,92 @@
...
+  }
 };
-export default props;
 
+
+export {
+  responses,
+  props,
+  expected
+};
```
Comment (Lumexralph) [inline client/src/tests/modules/auth/authMockData.js]: Good one! putting the mock data in a separate directory

**Review by Lumexralph — APPROVED** [review]

File: client/src/tests/modules/auth/container/index.test.jsx (hunk 3 above)
Comment (veeqtor) [reply client/src/tests/modules/auth/container/index.test.jsx]: Nice, I did not think about it in that aspect, I guess we will still achieve the same thing either way.

File: client/src/tests/modules/auth/container/index.test.jsx (hunk 1 above)
Comment (veeqtor) [reply client/src/tests/modules/auth/container/index.test.jsx]: Thanks for the suggestion, the reason behind that was that the state is not being set when testing for errors inside the `handleSuccess` method, therefore no need to check the state.

**Review by chidioguejiofor — APPROVED** [review]

**Review by Chieze-Franklin — APPROVED** [review]
LGTM!

**Pull request merged**